template/template.go

```diff
@@ -451,7 +451,9 @@ func (e *executer) resolvePreferences(m ast.Matcher, res []any) [][]string {
 				switch key := vKey.(type) {
 				case ast.CatchAllKey:
 					continue
-				case ast.NameLiteral, ast.QuotedLiteral, ast.NumberLiteral:
+				case ast.QuotedLiteral:
```

Comment:
Does MF2 expect raw string value?

Reply:
yes, it's true in the MF2 syntax context. But not logical processing context.

Reply:
The MF2WG test case expects `1` to be equal to `|1|` which means either quote unquoted literals, unquote quoted literals, or use custom comparison function (instead of `if key == value`)

Reply:
https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#pattern-selection

Reply:
> The method MatchSelectorKeys is determined by the implementation. It takes as arguments a resolved selector value rv and a list of string keys keys, and returns a list of string keys in preferential order.

Reply:
Ok, I see. We convert all literals to raw string values and then do a comparison.

In reality, we don't need to convert a number literal (float64) to string to compare. https://github.com/unicode-org/message-format-wg/blob/main/spec/formatting.md#example-3

We can resolve it separately. Not in this PR.